From e5070d1de828da662b9d33b0ce62962526e6db95 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 22 Jun 2020 07:16:04 +0000
Subject: [PATCH 1/8] net: ocelot: felix: register notifiers to use switchdev
 notification machanism

Kernel v5.2 replaces the port attr set functions with a notification
in switchdev:
  commit d45224d604c5 ("net: switchdev: Replace port attr set SDO with a notification")

And the ocelot has added relative ocelot_switchdev_nb notifier
and ocelot_switchdev_blocking_nb notifier to support switchdev
notification machanism:
  commit 56da64bc003b ("net: mscc: ocelot: Handle SWITCHDEV_PORT_ATTR_SET")

In ocelot/felix switch, we need to register these ocelot notifiers to
make the switch work normal.

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index dc6b8f64a504..cd522377b45d 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -798,6 +798,8 @@ static int felix_pci_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		goto err_ports_init;
 
 	register_netdevice_notifier(&ocelot_netdevice_nb);
+	register_switchdev_notifier(&ocelot_switchdev_nb);
+	register_switchdev_blocking_notifier(&ocelot_switchdev_blocking_nb);
 
 	dev_info(&pdev->dev, "%s v%s\n", FELIX_DRV_STR, FELIX_DRV_VER_STR);
 	return 0;
-- 
2.17.1

